jetty-ee9/jetty-ee9-maven-plugin/src/main/java/org/eclipse/jetty/ee9/maven/plugin/MavenWebInfConfiguration.java

```diff
@@ -51,15 +51,15 @@ public void configure(WebAppContext context) throws Exception
     {
         MavenWebAppContext jwac = (MavenWebAppContext)context;
 
-        //put the classes dir and all dependencies into the classpath
-        if (jwac.getClassPathFiles() != null && context.getClassLoader() instanceof WebAppClassLoader)
+        // put the classes dir and all dependencies into the classpath
+        if (jwac.getClassPathUris() != null && context.getClassLoader() instanceof WebAppClassLoader loader)
         {
             if (LOG.isDebugEnabled())
                 LOG.debug("Setting up classpath ...");
-            WebAppClassLoader loader = (WebAppClassLoader)context.getClassLoader();
-            for (File classpath : jwac.getClassPathFiles())
+
+            for (URI uri : jwac.getClassPathUris())
             {
-                loader.addClassPath(classpath.getCanonicalPath());
+                loader.addClassPath(uri.toASCIIString());
```

Comment:
Why are you using `toASCIIString` instead of `toString`?

Reply:
`toString()` returns the input string in most cases.

`toASCIIString()` returns a properly encoded string in all cases.